tests/test_starfile_stack.py

```diff
@@ -14,47 +14,43 @@
 
 
 class StarFileTestCase(TestCase):
-    def run(self, result=None):
-        """Overridden run method to use context manager provided by importlib_resources"""
-        with importlib_resources.path(
-            tests.saved_test_data, "sample_relion_data.star"
-        ) as path:
+    def setUpStarFile(self, starfile_name, mrcs_name):
+        # this method is run for both subclasses and creates a temporary directory to run tests
+        # as well as initializing a RelionSource
+        with importlib_resources.path(tests.saved_test_data, starfile_name) as path:
 
-            # Create a temporary file with the contents of the sample.mrcs file in a subfolder at the same location
+            # Create a temporary file with the contents of the .mrcs file in a subfolder at the same location
             # as the starfile, to allow our classes to do their job
-            temp_folder_path = os.path.join(path.parent.absolute(), "_temp")
+            self.temp_folder_path = os.path.join(path.parent.absolute(), "_temp")
```

Comment:
This is not a good way to make a temp folder.  Consider something like 

https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory

It may also reduce some of the other logic.